### PR TITLE
Registration and person entity

### PR DESCRIPTION
**Pull request by Cisfinitum**
- add Registration controller with regExp for email & password
- add some methods(like create, addToBlackList) to PersonDAO
- add some UnitTests to PersonDAOTest
- add intermediate methods to PersonDetailsServiceImpl & PersonService

### CODE REVIEW

File: src/main/java/com/epam/repository/PersonDAO.java [hunk 1]
```diff
@@ -32,4 +32,52 @@ Person buildPerson(ResultSet rs) throws SQLException {
...
 
+    public boolean doesEmailExists(String email) {
+        List<String> emailsList = jdbcTemplate.query("SELECT email FROM person", (rs, rowNum) -> getStringEmail(rs));
+        for(String stringEmail:  emailsList){
+            if (stringEmail.equals(email)) return true;
+        }
+        return false;
+    }
+
+    String getStringEmail(ResultSet rs) throws SQLException {
```
Comment (goponenkoev) [inline src/main/java/com/epam/repository/PersonDAO.java]: I think method name could be just `getEmail`

File: src/main/java/com/epam/repository/PersonDAO.java [hunk 2]
```diff
@@ -32,4 +32,52 @@ Person buildPerson(ResultSet rs) throws SQLException {
...
+        return rs.getString("email");
+    }
+
+    public boolean addPersonToDataBase(Person person) {
```
Comment (goponenkoev) [inline src/main/java/com/epam/repository/PersonDAO.java]: Please call methods simpler. Just addPerson. It's obvious we add it in DB if this method located in DAO

Comment (goponenkoev) [reply src/main/java/com/epam/repository/PersonDAO.java]: I think would be the better return Person, not boolean

File: src/main/java/com/epam/model/Person.java [hunk 3]
```diff
@@ -10,4 +10,10 @@
     private String email;
     private String password;
     private PersonRoleEnum role;
+
+    public Person(String email, String password, String role) {
```
Comment (ghost) [inline src/main/java/com/epam/model/Person.java]: You can try allArgsConstructor here

File: src/main/java/com/epam/service/PersonService.java [hunk 4]
```diff
@@ -24,4 +24,23 @@ public Person getPerson(String name){
         return null;
     }
 
+    public boolean addPersonToDataBase(Person person){
+        return personDAO.addPersonToDataBase(person);
+    }
+
+    public boolean updatePassword(String email, String password) {
```
Comment (ghost) [inline src/main/java/com/epam/service/PersonService.java:31]: Whats happened, if i pass empty string here?

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/model/Person.java (hunk 3 above)
Comment (Cisfinitum) [reply src/main/java/com/epam/model/Person.java]: The thing is that we should create an instance of Person in Controller. In all-args construct we must transfer to construct id, but we wont use it in any of other methods.

File: src/main/java/com/epam/service/PersonService.java (hunk 4 above)
Comment (Cisfinitum) [reply src/main/java/com/epam/service/PersonService.java:31]: I think that this case will not happen ever cuz when user wana to change password, we ll read all neccesary data from him (current user), but i added if-else constructor to method, wich instantly return false, if password || email == ""; But, in my opinion that is unnecessary cuz, we must check if password matches out password pattern(as i did in registration controller).

File: src/main/java/com/epam/repository/PersonDAO.java (hunk 1 above)
Comment (Cisfinitum) [reply src/main/java/com/epam/repository/PersonDAO.java]: Done

File: src/main/java/com/epam/repository/PersonDAO.java (hunk 2 above)
Comment (Cisfinitum) [reply src/main/java/com/epam/repository/PersonDAO.java]: Renamed
What about second request, i thought, that if adding is possible, and the operation completed sucesefully, we return true == "all good"

File: src/main/java/com/epam/model/Person.java (hunk 3 above)
Comment (Cisfinitum) [reply src/main/java/com/epam/model/Person.java]: Done by @RequairedArgsConstructor

File: src/main/webapp/WEB-INF/jsp/registration.jsp [hunk 5]
```diff
@@ -22,41 +22,38 @@
...
-                        <label for="first_name"><spring:message code="firstname" /></label>
-                    </div>
-                    <div class="input-field col s6">
-                        <input id="last_name" type="text" class="validate" name="secondname">
-                        <label for="last_name"><spring:message code="lastname" /></label>
+                        <input id="password2" type="password" class="validate" name="password2" onchange="checkPass()">
+                        <label for="password2"><spring:message code="passwordname" /></label>
                     </div>
                 </div>
+                <%--</div>--%>
```
Comment (goponenkoev) [inline src/main/webapp/WEB-INF/jsp/registration.jsp]: Is it commented code? We really need it?

Comment (Cisfinitum) [reply src/main/webapp/WEB-INF/jsp/registration.jsp]: changed

File: src/main/java/com/epam/repository/PersonDAO.java [hunk 6]
```diff
@@ -32,4 +32,54 @@ Person buildPerson(ResultSet rs) throws SQLException {
                 .build();
     }
 
+    public boolean doesEmailExists(String email) {
```
Comment (ghost) [inline src/main/java/com/epam/repository/PersonDAO.java:35]: without does, please

File: src/main/java/com/epam/service/PersonService.java (hunk 4 above)
Comment (ghost) [reply src/main/java/com/epam/service/PersonService.java:31]: but if i want to use your method from another service, and i will pass null here

**Review by ghost — CHANGES_REQUESTED** [review]

**Pull request merged**